tests/functional/curate/cram/format-dates/failure-reporting.t

```diff
@@ -29,6 +30,7 @@ This is expected to fail with an error, so redirecting stdout since we don't car
   ERROR: Unable to format dates for the following (record, field, date string):
   (0, 'collectionDate', '2020-01')
   (0, 'releaseDate', '2020-01')
+  Current expected date formats are ['%Y-%m-%d', '%Y-%m-XX', '%Y-XX-XX', 'XXXX-XX-XX', '%Y', '%Y-%m-%dT%H:%M:%SZ']. This can be updated with --expected-date-formats.
```

Comment:
Hmm, now that you mention it, I don't think this is clear in any of the help / docs. 

The default behavior for the `extend` action in argparse is to extend the defaults. We have a custom [ExtendOverwriteDefault](https://github.com/nextstrain/augur/blob/862aa37091203b718902cbbbd28d17452f6f9777/augur/argparse_.py#L81) action that overwrites the defaults instead of extending them, but this custom action does not change the output of the CLI help docs. 

Comparing `--metadata-delimiters` which uses `ExtendOverwriteDefault` and `--expected-date-formats` which uses the default `extend` action. Neither the help nor the docs differentiates the extend vs overwrite behavior:

```
--metadata-delimiters METADATA_DELIMITERS [METADATA_DELIMITERS ...]
                        Delimiters to accept when reading a plain text metadata file. Only one delimiter will be inferred. (default: (',', '\t'))
--expected-date-formats EXPECTED_DATE_FORMATS [EXPECTED_DATE_FORMATS ...]
                        Expected date formats that are currently in the provided date fields, defined by standard format codes as listed at
                        https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes. If a date string matches multiple formats, it will be parsed as the
                        first matched format in the provided order. (default: ['%Y-%m-%d', '%Y-%m-XX', '%Y-XX-XX', 'XXXX-XX-XX'])
```

![Screenshot 2024-10-17 at 12 17 10 PM](https://github.com/user-attachments/assets/dcadc7f4-8ba2-48d2-b141-c6b8dd4013f7)
![Screenshot 2024-10-17 at 12 17 21 PM](https://github.com/user-attachments/assets/90434d55-8283-4ef3-82aa-cfc49591e7e1)

Reply:
> [`--metadata-delimiters` cf. `--expected-date-formats`] Neither the help nor the docs differentiates the extend vs overwrite behavior

Perfect example right there!

Reply:
Tracking separately in https://github.com/nextstrain/augur/issues/1654